#!/usr/bin/env python
PACKAGE = "wego"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Red HSV thresholds (two ranges for red hue wrap-around)
gen.add("red_h_low1", int_t, 0, "Red H low 1", 0, 0, 180)
gen.add("red_h_high1", int_t, 0, "Red H high 1", 10, 0, 180)
gen.add("red_h_low2", int_t, 0, "Red H low 2", 160, 0, 180)
gen.add("red_h_high2", int_t, 0, "Red H high 2", 180, 0, 180)
gen.add("red_s_low", int_t, 0, "Red S low", 100, 0, 255)
gen.add("red_s_high", int_t, 0, "Red S high", 255, 0, 255)
gen.add("red_v_low", int_t, 0, "Red V low", 100, 0, 255)
gen.add("red_v_high", int_t, 0, "Red V high", 255, 0, 255)

# Green HSV thresholds
gen.add("green_h_low", int_t, 0, "Green H low", 35, 0, 180)
gen.add("green_h_high", int_t, 0, "Green H high", 85, 0, 180)
gen.add("green_s_low", int_t, 0, "Green S low", 100, 0, 255)
gen.add("green_s_high", int_t, 0, "Green S high", 255, 0, 255)
gen.add("green_v_low", int_t, 0, "Green V low", 100, 0, 255)
gen.add("green_v_high", int_t, 0, "Green V high", 255, 0, 255)

# Detection parameters
gen.add("min_area", int_t, 0, "Minimum detection area", 500, 0, 10000)
gen.add("circularity_thresh", double_t, 0, "Circularity threshold", 0.6, 0.0, 1.0)

# ROI settings (traffic light is at top of image)
gen.add("roi_top", int_t, 0, "ROI top", 0, 0, 480)
gen.add("roi_bottom", int_t, 0, "ROI bottom", 240, 0, 480)
gen.add("roi_left", int_t, 0, "ROI left", 0, 0, 640)
gen.add("roi_right", int_t, 0, "ROI right", 640, 0, 640)

# Brightness adjustment
gen.add("brightness_factor", double_t, 0, "Brightness factor (0.1-1.0, lower=darker)", 0.5, 0.1, 1.0)

exit(gen.generate(PACKAGE, "traffic_light_node", "TrafficLight"))
